### PR TITLE
Fixed bug with non null terminated string with atoi

### PR DESCRIPTION
**Pull request by Kswin01**
Null terminating `got_char` in virt rx.

### CODE REVIEW

**Comment by Ivan-Velickovic** [comment]
Separate the style commit and the commit that fixes the bug,

File: serial/components/virt_rx.c [hunk 1]
```diff
@@ -157,24 +157,26 @@ void handle_rx() {
         // This is for our RAW mode, char by char
         if (UART_MODE == RAW_MODE) {
             // microkit_dbg_puts("In raw mode virt rx\n");
-            char got_char = chars[0];
+            char got_char[2];
```
Comment (Ivan-Velickovic) [inline serial/components/virt_rx.c:164]: Doing comparisons on `got_char[0]` everywhere makes it a bit hard to read, instead, I would only define `char atoi_str[2]` when it's actually needed (which is not all cases). Not saying `atoi_str` is the right name but I think you know what I mean.

Comment (Kswin01) [reply serial/components/virt_rx.c]: That makes sense, I've added those changes now.

**Review by Ivan-Velickovic — APPROVED** [review]

File: serial/components/virt_rx.c [hunk 2]
```diff
@@ -174,7 +178,11 @@ void handle_rx() {
                 } else {
                     // Ensure that multi client input is off
                     multi_client = 0;
-                    int new_client = atoi(&got_char);
+                    // Null terminate got_char to be safe to use with atoi
+                    char got_char_terminated[2];
+                    got_char_terminated[0] = got_char;
+                    got_char_terminated[1] = '\0';
+                    int new_client = atoi(got_char_terminated);
```
Comment (Indanz) [inline serial/components/virt_rx.c:185]: `atoi()` is for converting strings, not single characters, please just use:

```c
int new_client = chars[0] - '0';
```

Comment (Ivan-Velickovic) [reply serial/components/virt_rx.c:185]: That won't work for more than 10 clients though, but neither does the current code.

All of this code will be revisited soon anyways with https://github.com/au-ts/sddf/issues/53. I've added a note about the use of `atoi` in the issue.

**Pull request merged**